consensus/misc/kip71.go

```diff
@@ -0,0 +1,74 @@
+package misc
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/math"
+	"github.com/klaytn/klaytn/params"
+)
+
+func VerifyKIP71Header(config *params.ChainConfig, parentHeader, header *types.Header) error {
+	if header.BaseFee == nil {
+		return fmt.Errorf("header is missing baseFee")
+	}
+	// Verify the baseFee is correct based on the parent header.
+	expectedBaseFee := CalcBaseFee(parentHeader, config)
+	if header.BaseFee.Cmp(expectedBaseFee) != 0 {
+		return fmt.Errorf("invalid baseFee: have %s, want %s, parentBaseFee %s, parentGasUsed %d",
+			header.BaseFee, expectedBaseFee, parentHeader.BaseFee, parentHeader.GasUsed)
+	}
+	return nil
+}
+
+func CalcBaseFee(parentHeader *types.Header, config *params.ChainConfig) *big.Int {
+	// If the current is the kip71 disabled block, then return default base fee (250ston)
+	if !config.IsKIP71ForkEnabled(parentHeader.Number) {
+		return new(big.Int).SetUint64(config.UnitPrice)
```

Comment:
I thought default base fee should be `0` for before fork. Is there any reason why default base fee is `UnitPrice`?

Reply:
- We'll call this function, `CalcBaseFee()` from worker and txPool with hard fork branch. In other words, It is impossible to call the function before hardfork. You can check the code in temporary branch for test (https://github.com/kjeom/klaytn/blob/dynamicGasPriceForTest/work/worker.go#L542).
- The following code is for the first hardfork block. We should set the base fee with nonzero like the `unitprice` because we need it for calculating next base fee.
```go
if !config.IsKIP71ForkEnabled(parentHeader.Number) {
		return new(big.Int).SetUint64(config.UnitPrice)
```